packages/Telephony/Common/RegexAggregatorInput.cs

```diff
@@ -0,0 +1,114 @@
+{
+    /// <summary>
+    /// Aggregates input until it matches a regex pattern and then stores the result in an output property.
+    /// </summary>
+    public class RegexAggregatorInput : Dialog
+    {
+        /// <summary>
+        /// Class identifier.
+        /// </summary>
+        [JsonProperty("$kind")]
+        public const string Kind = "Microsoft.Telephony.RegexAggregatorInput";
+        private const string AggregationDialogMemory = "dialog.aggregation";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RegexAggregatorInput"/> class.
+        /// </summary>
+        /// <param name="sourceFilePath">Optional, source file full path.</param>
+        /// <param name="sourceLineNumber">Optional, line number in source file.</param>
+        [JsonConstructor]
+        public RegexAggregatorInput([CallerFilePath] string sourceFilePath = "", [CallerLineNumber] int sourceLineNumber = 0)
+            : base()
+        {
+            // enable instances of this command as debug break point
+            this.RegisterSourceLocation(sourceFilePath, sourceLineNumber);
+        }
+
+        /// <summary>
+        /// Gets or sets the regex pattern to use to decide when the dialog has aggregated the whole message.
+        /// </summary>
+        [JsonProperty("terminationConditionRegexPattern")]
+        public StringExpression TerminationConditionRegexPattern { get; set; }
+
+        /// <summary>
+        /// Gets or sets the property to assign the result to.
+        /// </summary>
+        [JsonProperty("property")]
+        public StringExpression Property { get; set; }
+
+        /// <summary>
+        /// Gets or sets the activity to send to the user.
+        /// </summary>
+        /// <value>
+        /// An activity template.
+        /// </value>
+        [JsonProperty("prompt")]
+        public ITemplate<Activity> Prompt { get; set; }
+
+        /// <inheritdoc/>
+        public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc, object options = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            ITemplate<Activity> template = this.Prompt ?? throw new InvalidOperationException($"InputDialog is missing Prompt.");
+            IMessageActivity msg = await this.Prompt.BindAsync(dc, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+            if (msg != null && string.IsNullOrEmpty(msg.InputHint))
+            {
+                msg.InputHint = InputHints.ExpectingInput;
+            }
+
+            var properties = new Dictionary<string, string>()
+                {
+                    { "template", JsonConvert.SerializeObject(template) },
+                    { "result", msg == null ? string.Empty : JsonConvert.SerializeObject(msg, new JsonSerializerSettings() { NullValueHandling = NullValueHandling.Ignore }) },
+                };
+            TelemetryClient.TrackEvent("GeneratorResult", properties);
```

Comment:
This is cool. Is the TelemetryClient  available to all actions? Assume this is going to the customers app insights. Any rules for what can and cannot be logged.

Reply:
Sounds like an @carlosscastro question :D From what I see in the codebase it's pretty fast and loose.